0.10/s2i/bin/assemble

```diff
@@ -47,8 +47,51 @@ if [ -n "$NPM_MIRROR" ]; then
 	npm config set registry $NPM_MIRROR
 fi
 
-echo "---> Building your Node application from source"
-npm install
+# Set the DEV_MODE to false by default.
+if [ -z "$DEV_MODE" ]; then
+  export DEV_MODE=false
+fi
+
+# If NODE_ENV is not set by the user, then NODE_ENV is determined by whether
+# the container is run in development mode.
+if [ -z "$NODE_ENV" ]; then
+  if [ "$DEV_MODE" == true ]; then
+    export NODE_ENV=development
+  else
+    export NODE_ENV=production
+  fi
+fi
+
+if [ "$NODE_ENV" != "production" ]; then
+
+	echo "---> Building your Node application from source"
+	npm install
+
```

Comment:
Should `npm run build` not also be called here? For example, a Babel or TypeScript project will always need it, even in DEV.

Reply:
Hi @evanshortiss,

Another approach is to add a postinstall script in your `package.json`

```
  "scripts": {
     "postinstall": "npm run build"
```

Reply:
Yep, that's true, but I think we should have a standard approach in this file? Why call `npm run build` in PROD, but not DEV?

Reply:
> I think we should have a standard approach in this file
I fully agree with this.